### ISSUE TITLE
ESP32 WiFi Disconnect / Reconnect

### ISSUE THREAD

**Issue by vseven** [DanielOgorchock/ST_Anything#102]
So it turns out not only is the setAutoConnect(true) broke (which I think you knew) but the reason its broke it because it calls WiFi.Begin.   WiFi.Begin, if it doesn't instantly see the AP, clears out the SSID and password so sequential attempts fail.  I've been going back and forth, as have other people, on the ESP32 GitHub about this for almost three months now.  (https://github.com/espressif/arduino-esp32/issues/653) 

Long story short they said the only reliable way to do this is to use the following instead of WiFi.Begin:

WiFi.disconnect(true);
WiFi.begin(ssid, passwd);

per the post here:  https://github.com/espressif/arduino-esp32/issues/653#issuecomment-355659659 .  I was going to test this myself before putting in a pull request but wanted to give you a heads up.

**Comment by tagyoureit** [comment]
On this note, I see that the following was recently added:

> "Last Updated" date and time on all Child Devices;"

Is it possible to add a generic last update date/time to the master device?  A couple of my ESP8266 are in areas with sketchy WIFI.  I'd really like to have a notification (through WebCore) if the last time the ESP8266 sent an update (any update) to the SmartThings hub is >x minutes.  

Thanks!

**Comment by vseven** [comment]
Wouldn't it be easier to just look at one of the devices that is attached to the ESP8266?  If a device is updating then the controller is also updating and there is nothing else to do since each device already has that piece of information.

**Comment by tagyoureit** [comment]
The two devices I'm currently using are the relay shield (as a switch) and the motion sensor.  The switch will only update the status when it is told to do so, and the motion sensor will only send an update when it senses motion.  (This is my understanding, anyway).  So the switch won't be of use to let me know that the ESP8266 is still connected over wifi.  And it is impossible to tell the difference between a motion sensor that doesn't sense motion (no update sent = no change to date/time) and a device that is disconnected from the wifi.

I thought about using the RSSI value as an indicator, but the ST Anything platform just uses the last value received if communications stops.  

If there is a DHT shield (or the like) that continually sends data then yes, I could use that for sure, but it isn't in this setup.

Let me know if there is something I'm missing,  but I think the only way is to have something along the lines of "last communication with master device".  Last update time on the RSSI value (or a new tile) would likely suffice for me.

**Comment by vseven** [comment]
Hrm.  I know I have a contact sensor on my garage heater and I get a update every so many minutes even if the heater is physically turned off and doesn't change state, just a update saying its still off.  The motion sensor is the within the same class so I thought it did the same thing.  Have you verified the last update time on your motion sensor?  It should be updating even without motion.

**Comment by ogiewon** [comment]
By default, all devices refresh their status every 5 minutes, even if nothing has changed.

**Comment by tagyoureit** [comment]
Ah, ok.  So long as the update time keeps changing, I can assume there is a connection.  If the last update time stops updating (after 5 minutes), then it should be considered that the connection is broken.  That makes sense.  Thanks for clarifying.

**Comment by vseven** [comment]
I would even wait for the second poll, it's possible a intermittent condition just caused the last send to fail.  Personally knowing its updating at least every 5 minutes I'd look for a time stamp older then say 12 minutes.  That means you missed two consecutive updates and maybe something is wrong.

As for the reason this issue was opened in the first place I still ahven't had a chance to test the changes.....the auto reboot logic is working too well and when my ESP32 can't reconnect it just reboots and all is good.  If this were doing lighting control (another project I haven't finished) it would be a big deal.  I will figure it out though.

**Comment by tagyoureit** [comment]
Agreed on the poll.  And I don't even need that granularity.  Even 30 or 60 minutes would work for what I want.  Thanks again.
